tools/agent_tool/src/cpp/main.cpp

```diff
@@ -0,0 +1,245 @@
+
+#include "user_interface/arguments_configuration.h"
+
+enum EntityType
+{
+    CLIENT,
+    SERVER,
+    REPEATER
+};
+
+int main(
+        int argc,
+        char** argv)
+{
+    eprosima::utils::event::SignalEventHandler<eprosima::utils::event::Signal::sigint> sigint_handler;
+
+    // Help message formatting settings
+    int columns;
+
+#if defined(_WIN32)
+    char* buf = nullptr;
+    size_t sz = 0;
+    if (_dupenv_s(&buf, &sz, "COLUMNS") == 0 && buf != nullptr)
+    {
+        columns = strtol(buf, nullptr, 10);
+        free(buf);
+    }
+    else
+    {
+        columns = 80;
+    }
+#else
+    columns = getenv("COLUMNS") ? atoi(getenv("COLUMNS")) : 80;
+#endif // if defined(_WIN32)
+
+    // Parameter definition
+    EntityType entity_type = EntityType::CLIENT;
+    std::string connection_address = "127.0.0.1";
+    std::string listening_address = "127.0.0.1";
+    std::string ip;
+    int domain = 0;
+    int connection_port = 12121;
+    int listening_port = 12121;
+    int port;
+    char* name = "agent_tool";
+    eprosima::ddsrouter::core::types::TransportProtocol transport_protocol =
+            eprosima::ddsrouter::core::types::TransportProtocol::udp;
+
+    // Parse example options
+    argc -= (argc > 0);
+    argv += (argc > 0); // skip program name argv[0] if present
+    option::Stats stats(usage, argc, argv);
+    std::vector<option::Option> options(stats.options_max);
+    std::vector<option::Option> buffer(stats.buffer_max);
+    option::Parser parse(usage, argc, argv, &options[0], &buffer[0]);
+
+    if (parse.error())
+    {
+        option::printUsage(fwrite, stdout, usage, columns);
+        return 1;
+    }
+
+    if (options[optionIndex::HELP])
+    {
+        option::printUsage(fwrite, stdout, usage, columns);
+        return 0;
+    }
+
+    for (int i = 0; i < parse.optionsCount(); ++i)
+    {
+        option::Option& opt = buffer[i];
+        switch (opt.index())
+        {
+            case optionIndex::HELP:
+                // Not possible as this argument has been already handled and it exits the program
+                break;
+
+            case optionIndex::ENTITY_TYPE:
+                if (strcmp(opt.arg, CLIENT_ENTITY_KIND_ARG) == 0)
+                {
+                    entity_type = EntityType::CLIENT;
+                }
+                else if (strcmp(opt.arg, SERVER_ENTITY_KIND_ARG) == 0)
+                {
+                    entity_type = EntityType::SERVER;
+                }
+                else if (strcmp(opt.arg, REPEATER_ENTITY_KIND_ARG) == 0)
+                {
+                    entity_type = EntityType::REPEATER;
+                }
+                else
+                {
+                    std::cerr << "ERROR: incorrect entity type. Only <client|server|repeater> accepted." << std::endl;
+                    return 1;
+                }
+                break;
+
+            case optionIndex::CONNECTION_ADDRESS:
+                connection_address = std::string(opt.arg);
```

Comment:
As it is done now, it would be better to write directly in `ip` variable. 
The idea here is to add 2 different arguments so user API is kept intact in future versions (and thus user experience improved)